### PR TITLE
mention additional librairies

### PR DESCRIPTION
**Pull request by OlivierBondu**
## What changes are proposed in this pull request?

Mention early on that optional librairies may be requiredif one goes beyond the basic setup (to avoid ugly crashes at runtime)
This PR is the follow-up to the conversation here: https://github.com/mlflow/mlflow/issues/2837. As discussed in said Issue, it may be interesting to comment properly [this requirements.txt](https://github.com/mlflow/mlflow/blob/master/travis/large-requirements.txt) which I think is the most complete one that anyone should ever need ?

## How is this patch tested?

Documentation - no test done

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Mention early on that optional librairies may be requiredif one goes beyond the basic setup (to avoid ugly crashes at runtime)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/quickstart.rst
Comment (dbczumar) [inline docs/source/quickstart.rst:28]:
```suggestion
In order to use some of MLflow features (ML librairies, storage options, ...), you may need to install extra libraries.
```

Comment (dbczumar) [reply docs/source/quickstart.rst:28]: Can we provide an example? E.g., "For example, the `mlflow.tensorflow` module requires TensorFlow to be installed."

File: docs/source/quickstart.rst
Comment (dbczumar) [inline docs/source/quickstart.rst:30]:
```suggestion
Please refer to this `requirements.txt <https://github.com/mlflow/mlflow/blob/master/travis/large-requirements.txt>`_ for a complete list of dependencies.
```

^ Some of the dependencies in this file are required (e.g., `sqlalchemy`)

**Review by dbczumar — COMMENTED** [review]
@OlivierBondu This is great! Left a few small suggestions; excited to merge this once they're addressed.

File: docs/source/quickstart.rst
Comment (OlivierBondu) [reply docs/source/quickstart.rst]: excellent idea :)

**Comment by OlivierBondu** [comment]
PR updated

**Review by dbczumar — APPROVED** [review]
@OlivierBondu Sorry for the delay here. This LGTM! In order to merge this PR, can you follow the PR signoff instructions for DCO compliance? https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#sign-your-work

**Comment by OlivierBondu** [comment]
Thanks, I just did sign off the commits, hopefully this will work.

**Comment by OlivierBondu** [comment]
mmm, please do not merge yet, there is some mixup in the commit signoffs...

**Comment by OlivierBondu** [comment]
Should finally be fixed :)
Sorry @dbczumar I had to 'signoff' your last commit. I am not _that_ much of a git master, and could not find out how to have different author, committer and sign-off, so I just aligned the 3 to my own name and email, I hope you don't mind.

**Comment by OlivierBondu** [comment]
ping !

**Pull request merged**
